### PR TITLE
MRG: rework the manifest documentation; do misc cleanup

### LINKED ISSUE #3009: `sourmash sig manifest` on zip files yields unusable standalone manifest CSVs
The `internal_location` is kept as the internal location in the zipfile, rather than the name of the zipfile:

```
% sourmash sig manifest example.sig.zip -o mf.csv
...
% cat mf.csv | cut -c, -f1
# SOURMASH-MANIFEST-VERSION: 1.0
internal_location
signatures/324074c7287ed934af4fd0a6a459aa30.sig.gz
% sourmash sig describe mf.csv
...
ERROR: Error while reading signatures from 'signatures/324074c7287ed934af4fd0a6a459aa30.sig.gz'.
```

### LINKED ISSUE #3023: how to support lazy/streaming load with standard sourmash functions: standalone manifests
@anneliekth asked me yesterday how to best provide a list of metagenome sketches to `mgmanysearch` (see https://github.com/sourmash-bio/sourmash_plugin_containment_search/). I realized I wasn't 100% sure of the answer, despite having written [this](https://github.com/sourmash-bio/sourmash_plugin_branchwater/blob/main/doc/README.md):

>so the question becomes, how do you provide collections of large metagenomes to manysearch and fastmultigather in a single filename?
>And the answer is: manifests. Manifests are a sourmash filetype that contains information about sketches without containing the actual sketch content, and they can be used as "catalogs" of sketch content.

(Part of my confusion was that the text above is being used through Rust functionality, not through standard Python loading functions.)

`mgmanysearch` uses standard sourmash loading functions, so I thought an investigation would be useful and lead to some add'l sourmash documentation too!

tl;dr don't use pathlists, use manifests.

## the script

I wrote the following Python script:
```python
#! /usr/bin/env python
import sys
import sourmash
import time

print(f'opening {sys.argv[1]}')
sys.stdout.flush()
mark = time.time()
idx = sourmash.load_file_as_index(sys.argv[1])
print(f'{time.time() - mark:.3f}s')
sys.stdout.flush()

print(f'selecting {sys.argv[1]}')
sys.stdout.flush()
mark = time.time()
idx = idx.select(ksize=21)
print(f'{time.time() - mark:.3f}s')
sys.stdout.flush()

print("starting...")
sys.stdout.flush()

mark = time.time()
for ss in idx.signatures():
    print(f'loaded {ss.name}')
    print(f'{time.time() - mark:.3f}s')
    sys.stdout.flush()
    mark = time.time()

```

## the execution

and then ran it on a pathlist containing a list of filenames:
```
opening pathlist.txt
23.715s
selecting pathlist.txt
0.000s
starting...
loaded 139_2
0.000s
loaded 139_1
0.000s
loaded 139_3
0.000s
loaded 139_4
0.000s
```

and on a manifest generated with `sourmash sig collect $(cat pathlist.txt) -o mf.csv  -F csv`
```
opening mf.csv
0.009s
selecting mf.csv
0.000s
starting...
loaded 139_1
4.404s
loaded 139_2
6.663s
loaded 139_3
5.751s
loaded 139_4
6.798s
```

## results

When using pathlists, all sketches are loaded at once at the beginning, consuming All The Memory.

When using manifests, all sketches are loaded on demand, **not** consuming All the Memory.

## other thoughts

This is another reason to use .zip files to store sketches, instead of sig.gz files; `sig collect` will need to load the actual sketches in sig.gz files in order to build the manifest, while the manifest is already available in .zip files.

## tl;dr

* if you have a bunch of big metagenomes to search using (e.g.) `mgmanysearch`,
* and you want to make them into a list to search,
* store them in zip files,
* and use `sig collect` to build a manifest across some or all of them,
* and then use those manifests.

TODO: verify that `sig collect` loads things on the command line progressively 😅 

Related issues:
* https://github.com/sourmash-bio/sourmash/issues/1899
* https://github.com/sourmash-bio/sourmash_plugin_branchwater/pull/197#issuecomment-1940337725
* https://github.com/sourmash-bio/sourmash/issues/1895

### LINKED ISSUE #3039: `sig collect` should resolve to individual files when loading a pathlist or a directory
suppose you have a pathlist containing a bunch of zip files, and you run `sig collect` directly on the pathlist:
```
sourmash sig collect podar-ref-zip-list.txt -o podar-ref-zip-list.mf.csv -F csv
```
you end up with a manifest CSV file that has `podar-ref-zip-list.txt` for all of the `internal_location` values.

if, instead, you do:
```
sourmash sig collect $(cat podar-ref-zip-list.txt) -o podar-ref-zip-list.mf.csv -F csv
```
you end up with the individual files going into the `internal_location` values, which is preferable.

this is because `sig collect` overwrites the "true" location of sketches with the location they were loaded from on the command line. This is appropriate in many other situations but is not great when loading from directories or pathlists or (I suspect) standalone manifests.

related issues:
* https://github.com/sourmash-bio/sourmash/issues/1895

### LINKED ISSUE #3040: inconsistencies in pathlist and directory loading with `MultiIndex`
ohmigod this is killing me... 😭 

When loading files, pathlists and directories in sourmash, we use `MultiIndex`.

For loading a pathlist, we use `MultiIndex.load_from_pathlist`. This will load any kind of index object from a text file containing a list of paths.

When loading a path, we use `MultiIndex.load_from_path`.
* if the path is a directory, we call `MultiIndex.load_from_directory`. This does a traversal of the directory hierarchy and tries to load all files using a `LinearIndex`, i.e. as a sig/sig.gz file.
* if the path is a single file that is a sig/sig.gz file, we load it using `LinearIndex`.

In particular this means that we can load all kinds of things (zip files, in particular, but also SBTs, LCAs, SQLite databases, or really anything for which we have a plugin) from pathlists. From directory hierarchies, however, we can only load sig/sig.gz files.

This seems frustrating and inconsistent.

Related:
* #3039

### LINKED ISSUE #3048: can we turn picklists + collections into manifests?
along the theme of cool ways to subset collections with manifests, @AnneliektH is converting `fastgather` results into manifests [here](https://github.com/AnneliektH/2023-swine-sra/blob/main/sourmash/Snakefile_mgmany). This is presumably because `mgmanysearch` doesn't support picklists (for which fastgather output could then be used), but _does_ supports standalone manifests.

so the question du jour is: do we have a standard way to go from picklists + collections => standalone manifest?

I think `sourmash sig check` might do it: [docs](https://sourmash.readthedocs.io/en/latest/command-line.html#sourmash-signature-check-compare-picklists-and-manifests). I will check and then recommend it to annie if so :).

it looks like `sourmash sig collect` does not, however: [docs](https://sourmash.readthedocs.io/en/latest/command-line.html#sourmash-signature-collect-collect-manifests-across-databases). That was my first guess.

related:
* https://github.com/sourmash-bio/sourmash/issues/3023
* #3027

### PR DESCRIPTION
**Pull request by ctb**
This PR:
* fixes a minor nit in `sourmash sig collect` output where it said "loaded 0 signatures"
* updates a lot of the documentation around standalone manifests to encourage their use
* in tandem, modifies docs to discourage loading from pathlists/from-files and directory hierarchies

TODO:
- [x] look at TODO item re directories in sig collect
- [x] think about adding https://github.com/sourmash-bio/sourmash/issues/3023 information into docs about lazy loading; maybe in the advanced databases document?
- [x] update `sig manifest` docs to point out that they do not generate standalone manifests
- [x] revisit branchwater plugin documentation to, to either make issues or make changes
- [x] update `sig check` and `sig collect` to tell people to expand their paths ref https://github.com/sourmash-bio/sourmash/issues/3039
- [x] update docs more to recommend against pathlists and directories per https://github.com/sourmash-bio/sourmash/issues/3040

Related issues:
* https://github.com/sourmash-bio/sourmash_plugin_branchwater/issues/235
* Fixes https://github.com/sourmash-bio/sourmash/issues/3048
* Fixes https://github.com/sourmash-bio/sourmash/issues/3009 by recommending `sig collect` and `sig check` instead of `sig manifest` for making standalone manifests
* https://github.com/sourmash-bio/sourmash/issues/3053
* Fixes https://github.com/sourmash-bio/sourmash/issues/3023
* Fixes https://github.com/sourmash-bio/sourmash/issues/3039
* Fixes https://github.com/sourmash-bio/sourmash/issues/3040

### CODE REVIEW

**Comment by ctb** [comment]
Ready for review & merge, @sourmash-bio/devs !

File: doc/command-line.md [hunk 1]
```diff
@@ -2092,7 +2114,7 @@ The following `coltype`s are currently supported for picklists:
 * `gather` - use the CSV output of `sourmash gather` as a picklist
 * `prefetch` - use the CSV output of `sourmash prefetch` as a picklist
 * `search` - use the CSV output of `sourmash prefetch` as a picklist
-* `manifest` - use the CSV output of `sourmash sig manifest` as a picklist
+* `manifest` - use CSV manifests produced by `sig manfiest` as a picklist
```
Comment (bluegenes) [inline doc/command-line.md]:
```suggestion
* `manifest` - use CSV manifests produced by `sig manifest` as a picklist
```

do we want to say `sig collect`/`sig check` here instead, if `sig manifest` is not the best way to produce standalone manifests?

**Review by bluegenes — APPROVED** [review]
suggested a typo fix, otherwise lgtm

File: doc/command-line.md (hunk 1 above)
Comment (ctb) [reply doc/command-line.md]: funny story, you can use `sig manifest` output as a picklist just fine 😆
